lib/redirect-logs.sh

```diff
@@ -1,19 +1,28 @@
 #!/usr/bin/env bash
 
-export DD_LOGS_CONFIG_TCP_FORWARD_PORT="${DD_LOGS_CONFIG_TCP_FORWARD_PORT:-10514}"
+export DD_LOGS_ENABLED="${DD_LOGS_ENABLED:-false}"
 export DISABLE_STD_LOG_COLLECTION="${DISABLE_STD_LOG_COLLECTION:-false}"
+export DD_LOGS_CONFIG_CUSTOM_CONFIG
+export DD_LOGS_CONFIG_TCP_FORWARD_PORT
 
+# redirect forwards all standard inputs to a TCP socket listening on port DD_LOGS_CONFIG_TCP_FORWARD_PORT.
 redirect() {
   while true; do
     nc localhost $DD_LOGS_CONFIG_TCP_FORWARD_PORT || true
   done
 }
 
+# setup the redirection from stdout/stderr to the logs-agent.
 if [ "$DD_LOGS_ENABLED" = "true" ]; then
-  if [ "$DISABLE_STD_LOG_COLLECTION" != "true" ]; then
-    echo "collect all logs forwarded to stdout/stderr"
-    exec &> >(tee >(redirect))
+  if [ -z "$DD_LOGS_CONFIG_TCP_FORWARD_PORT" ]; then
+    echo "can't collect logs, DD_LOGS_CONFIG_TCP_FORWARD_PORT is not set"
```

Comment:
Not sure about what's happening here, do we really need both `DD_LOGS_CONFIG_TCP_FORWARD_PORT` and `DD_LOGS_CONFIG_CUSTOM_CONFIG` ? looks like the configs will collide and try to create 2 TCPListener on 10514

don't we want something like (not sure of the exact syntax)

```
if [ -z "$DD_LOGS_CONFIG_TCP_FORWARD_PORT" || -z "DD_LOGS_CONFIG_CUSTOM_CONFIG" ]; then
  echo "can't collect..."
else
  echo collecting  ...
fi

Reply:
+1 for `if [ -z "$DD_LOGS_CONFIG_TCP_FORWARD_PORT" || -z "DD_LOGS_CONFIG_CUSTOM_CONFIG" ]; then`